### ISSUE TITLE
Building DEB packages fails

### ISSUE THREAD

**Issue by cRaZy-bisCuiT** [UltraStar-Deluxe/USDX#525]
### Actual behaviour

Trying to build DEB packages after a successfull build of USDX fails.

### Expected behaviour
The packaging should just be running finde producing some DEBs to install USDX witrh.

### Steps to reproduce

```
# check for missing dependencies
 dpkg-checkbuilddeps
# install missing dependencies
sudo apt -y install liblua5.3-dev libportmidi-dev libopencv-dev
trying to build the DEBs
dpkg-buildpackage -rfakeroot -b -uc
```
The following error appears:
```
# successfull build until here... then ->
Assembling uaudiodecoder_ffmpeg
Compiling media/UMedia_dummy.pas
Assembling umedia_dummy
Compiling menu/UMenuStaticList.pas
Assembling umenustaticlist
Assembling ultrastardx
Linking ../game/ultrastardx
ultrastardx.dpr(434) Warning: "crtbeginS.o" not found, this will probably cause a linking failure
ultrastardx.dpr(434) Warning: "crtendS.o" not found, this will probably cause a linking failure
/usr/bin/ld.bfd: warning: ../game/link.res contains output sections; did you forget -T?
/usr/bin/ld.bfd: lib/projectM/cwrapper/libprojectM-cwrapper.a(projectM-cwrapper.o): in function `__static_initialization_and_destruction_0':
/usr/include/libprojectM/Common.hpp:143: undefined reference to `__dso_handle'
/usr/bin/ld.bfd: ../game/ultrastardx: hidden symbol `__dso_handle' isn't defined
/usr/bin/ld.bfd: final link failed: bad value
ultrastardx.dpr(434) Error: Error while linking
ultrastardx.dpr(434) Fatal: There were 1 errors compiling module, stopping
Fatal: Compilation aborted
Error: /usr/bin/ppcarm returned an error exitcode
make[2]: *** [Makefile:245: ../game/ultrastardx] Error 1
make[2]: Leaving directory '/home/pi/Downloads/USDX/src'
make[1]: *** [Makefile:126: all] Error 2
make[1]: Leaving directory '/home/pi/Downloads/USDX'
dh_auto_build: make -j4 returned exit code 2
make: *** [debian/rules:13: binary] Error 2
dpkg-buildpackage: error: fakeroot debian/rules binary subprocess returned exit status 2
```
contents of debian/rules:13:
`dh $@ --with autotools_dev,autoreconf`

Is the build setup outdated? I never build DEBs on my own before, so it might just be me... 

- USDX version: git latest
- Operating System + version: Raspian Debian 10 Buster on RPI4

**Comment by s09bQ5** [comment]
Your distro messed up and didn't update fpc.cfg to point to the directory containing crtbeginS.o and crtendS.o for the currently installed GCC version.

**Comment by basisbit** [comment]
would be awesome to have TravisCI use these docker builds to build & upload these packages somewhere

**Comment by cRaZy-bisCuiT** [comment]
Thank you very much. The test here did not take place in Docker but on a bare metal Raspian. I'll try again in my docker environment and report back.

I never used Travis CI / created builds with that. If Docker runs well I could look into that / you guys could help me figure that out maybe.

**Comment by basisbit** [comment]
UltraStar uses TravisCi + Docker, so you could just look at the configuration file there. It is quite straightforward set up: https://github.com/UltraStar-Deluxe/Play/blob/master/.travis.yml

**Comment by cRaZy-bisCuiT** [comment]
Do you have a build server for Travis? Does Travis also take care of cross-compiling?

My setup right now needs arm-hardware to compile. The whole thing would need to wrapped up by something like dockercross to make it compile on x68_64.

**Comment by s09bQ5** [comment]
The Github repository contains a `.travis.yml` file. Everytime we push a branch or make a pull request Travis is informed and will use the instructions in that file to perform a bunch of builds. With Travis you don't have direct access to a fixed server. They pick a random machine to run the build jobs.

We don't do cross compiles. There is no special tooling for that in Travis, but nobody will stop us from installing a cross toolchain. They do have ARM64 machines, so we don't need to cross compile there.

**Comment by cRaZy-bisCuiT** [comment]
Well, right now we need x32 ARM. ARM64 will only be available in the future for RPI4/Raspian. I assume the performance will be better with that, also some changes to the codecs might be available for ffmpeg by then to make use of RPI4 hardware.

Still waiting for my Docker build to finish. I don't know yet if the issue is resolved. Building ffmpeg on a pi takes like 2 hours if you include a bunch of build flags.

**Comment by cRaZy-bisCuiT** [comment]
Well, the same error happens in the Docker build. Do you have any ideas how I could fix these issues? Would I have to rewrite fpc.cfg myself?

**Comment by s09bQ5** [comment]
On my Debian Buster /etc/fpc.cfg contains a block

```
# path to the gcclib
#ifdef cpui386
-Fl/usr/lib/gcc/x86_64-linux-gnu/8
#endif
#ifdef cpux86_64
-Fl/usr/lib/gcc/x86_64-linux-gnu/8
#endif
```

/usr/lib/gcc/x86_64-linux-gnu/8 is the directory containing crtbeginS.o and crtendS.o.
You need an -Fl line similar to that one in your fpc.cfg.

/etc/fpc.conf should have been updated by the postinst script in the fp-compiler package. Was gcc installed or updated after fpc had been installed?

**Comment by cRaZy-bisCuiT** [comment]
Thank you.

The build runs fine on x86 (using ffmpeg binary instead of compiling) ... USDX build finishes as well as DEB packaging. Looking into that next week, since I don't have my Pi with me this weekend.

**Comment by cRaZy-bisCuiT** [comment]
Rename the attached document and place it at:
`/etc/fpc.cfg`
This config works for Raspian Buster (May 2020 release) on armhf (32bit) RPI4.
[fpc.cfg.txt](https://github.com/UltraStar-Deluxe/USDX/files/4755580/fpc.cfg.txt)

**Comment by dgruss** [comment]
building on raspbian works completely fine (not deb but in general)

**Comment by barbeque-squared** [comment]
Yeah I think it's fair to close this. If the _building_ works but not some Debian-specific packaging thing, that sounds more like a Debian issue. Closing this as "Won't fix" for now. Maybe something is broken in our `debian` folder too, but when the distro itself starts shipping wrong /etc files, there's really not much we can do.
